exporter/azure_monitor/azure_monitor.go

```diff
@@ -37,17 +38,87 @@ var _ trace.Exporter = (*AzureTraceExporter)(nil)
 	@param sd Span data retrieved by opencensus
 */
 func (exporter *AzureTraceExporter) ExportSpan(sd *trace.SpanData) {
-	baseObj := common.BaseObject {IKey : exporter.options.InstrumentationKey}
 	envelope := common.Envelope {
-		BaseObject: baseObj,
+		IKey : exporter.Options.InstrumentationKey,
+		Tags : common.Azure_monitor_contect,
+		Name : "Microsoft.ApplicationInsights.RemoteDependency",
+		Time : getCurrentTime(),
+	}
+
+	if sd.ParentSpanID.String() == "" { 
+		// TODO: Add parent span details if any
+		fmt.Println("ADD PARENT DETAILS")
+		//envelope.Tags["ai.operation.parentId"] = 
+	}
+	if sd.SpanKind == trace.SpanKindServer {
+		// TODO: Add for server case
+		fmt.Println("ADD SERVER CASE")
+		envelope.Name = "Microsoft.ApplicationInsights.Request"
+	} else {
+		envelope.Name = "Microsoft.ApplicationInsights.RemoteDependency"
+		currentData := common.RemoteDependency{
+			Name : sd.Name,
+			Id : "|" + sd.SpanContext.TraceID.String() + "." + sd.SpanID.String() + ".",
+			ResultCode : "0", // TODO: Out of scope for now
+			Duration : timeStampToDuration(sd.EndTime.Sub(sd.StartTime)),
+			Success : true,
+			Ver : 2,
+		}
+		if sd.SpanKind == trace.SpanKindClient {
+			// TODO: Add for client case
+			fmt.Println("ADD CLIENT CASE")
+		} else {
+			currentData.Type = "INPROC" 
+		}
+		envelope.DataToSend = common.Data {
+			BaseData : currentData,
+			BaseType : "RemoteDependencyData",
+		}
 	}
-	envelope.Name = "Microsoft.ApplicationInsights.RemoteDependency"
 
 	transporter := common.Transporter{ 
 		EnvelopeData: envelope,
 	}
-	transporter.Transmit(&exporter.options, &envelope)
+	transporter.Transmit(&exporter.Options, &envelope)
 
 	fmt.Printf("Name: %s\nTraceID: %x\nSpanID: %x\nParentSpanID: %x\nStartTime: %s\nEndTime: %s\nAnnotations: %+v\n\n",
 		sd.Name, sd.TraceID, sd.SpanID, sd.ParentSpanID, sd.StartTime, sd.EndTime, sd.Annotations)
 }
+
+/* Generates the current time stamp and properly formats to a string.
+	@return time stamp
+*/
+func getCurrentTime() string {
+	t := time.Now()
+	formattedTime := t.Format("2006-01-02T15:04:05.000000Z")

```

Comment:
would be helpful to give some hint on the magic value.

Reply:
I used this website to help me format the date (http://gotime.agardner.me/). It allowed me to custom build a format of time that go could understand. All custom formats for go have to be of the time stamp Jan 2 15:04:05 2006 MST as mentioned here (https://godoc.org/time#Time.Format) and here more explicitly (https://stackoverflow.com/questions/50784043/define-custom-time-format).

Reply:
From the doc: "an easy way to remember this value is that it holds, when presented in this order, the values 1, 2, 3, 4, 5, 6, -7"